### PR TITLE
feat: add requests and responses to get lengths for sets, sorted sets and dictionary

### PR DESCRIPTION
**Pull request by pratik151192**
Added requests and responses to obtain lengths for set, sorted set, sorted set length by score, and dictionary.

### CODE REVIEW

File: momento/sorted_set_length_by_score.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/momentohq/client-sdk-go/responses"
+
+	pb "github.com/momentohq/client-sdk-go/internal/protos"
+)
+
+type SortedSetLengthByScoreRequest struct {
+	CacheName string
+	SetName   string
+	MinScore  *float64
+	MaxScore  *float64
```
Comment (cprice404) [inline momento/sorted_set_length_by_score.go:15]: I presume this is the same way we were exposing optional params for existing APIs like FetchByScore?

**Review by cprice404 — DISMISSED** [review]
lgtm but @pgautier404 should review.

one thing to keep in mind - if we merge this PR without the corresponding implementation, then we sort of can't do a release of the Go SDK off of the main branch until the implementation comes in. Just something to keep in mind in case an urgent bug fix elsewhere was needed. Not a big deal as long as the implementation PR is merged shortly after this one.

File: momento/sorted_set_length_by_score.go (hunk 1 above)
Comment (pratik151192) [reply momento/sorted_set_length_by_score.go:15]: that's correct.

**Comment by pratik151192** [comment]
Note that this is blocked on https://github.com/momentohq/client-sdk-go/pull/332#pullrequestreview-1515153029 anyway

**Comment by pratik151192** [comment]
This PR has become a bit clumsy due to a bunch of merges and rebases. I'm going to discard this and cut a new one.

**Pull request closed without merging**